src/lecture_feedback/thread_safe_dict.py

```diff
@@ -1,32 +1,50 @@
+from __future__ import annotations
+
 import threading
-from collections.abc import Iterator
-from typing import Any, Self
+from collections import UserDict
+from typing import TYPE_CHECKING, Any, Self
+
+if TYPE_CHECKING:
```

Comment:
why this condition?

Reply:
https://docs.astral.sh/ruff/rules/typing-only-standard-library-import/